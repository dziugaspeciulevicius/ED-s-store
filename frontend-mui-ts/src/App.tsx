import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import HomePage from "./pages/HomePage";
import ProductPage from "./pages/ProductPage";
import { ThemeProvider } from "@emotion/react";
import theme from "./theme";
import RoutingConstants from "./core/constants/routing";
import ResponsiveAppBar from "./components/ResponsiveAppBar";

const App = () => {
  return (
    <ThemeProvider theme={theme}>
      <div>
        <ResponsiveAppBar />
        <Router>
          <main>
            <Routes>
              <Route path={RoutingConstants.homepage} element={<HomePage />} />
              <Route
                path={RoutingConstants.product}
                element={<ProductPage />}
              />
            </Routes>
          </main>
        </Router>
      </div>
    </ThemeProvider>
  );
};

export default App;
