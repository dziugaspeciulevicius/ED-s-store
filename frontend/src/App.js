import React from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";

import Header from "./components/Header.component";
import Footer from "./components/Footer.component";
import HomePage from "./pages/HomePage.component";
import ProductPage from "./pages/ProductPage.component";
import ShopPage from "./pages/ShopPage.component";
import ContactPage from "./pages/ContactPage.component";
import AboutPage from "./pages/AboutPage.component";
import CartPage from "./pages/CartPage.component";
import LoginPage from "./pages/LoginPage.component";
import RegisterPage from "./pages/RegisterPage.component";
import DashboardPage from "./pages/DashboardPage.component";
import ShippingPage from "./pages/ShippingPage.component";
import PaymentPage from "./pages/PaymentPage.component";
import PlaceOrderPage from "./pages/PlaceOrderPage.component";
import OrderPage from "./pages/OrderPage.component";
import ScrollToTop from "./components/ScrollToTop.component";
import AdminPanel from "./pages/AdminPanel.component";
import UserEditPage from "./pages/UserEditPage.component";
import ProductEditPage from "./pages/ProductEditPage.component";

import NotFound from "./components/NotFound.component";

const App = () => {
  return (
    <div>
      <Router>
        <ScrollToTop />
        <Header />
        <main className="pb-3">
          <Switch>
            <Route exact path="/" component={HomePage} />
            <Route exact path="/page/:pageNumber" component={HomePage} />
            <Route
              exact
              path="/search/:keyword/page/:pageNumber"
              component={HomePage}
            />
            <Route exact path="/search/:keyword" component={HomePage} />
            <Route path="/product/:id" component={ProductPage} />
            <Route path="/cart/:id?" component={CartPage} />
            <Route path="/shop" component={ShopPage} />
            <Route path="/contact" component={ContactPage} />
            <Route path="/about" component={AboutPage} />
            <Route path="/login" component={LoginPage} />
            <Route path="/register" component={RegisterPage} />
            <Route path="/profile" component={DashboardPage} />
            <Route exact path="/admin" component={AdminPanel} />
            <Route exact path="/admin/user/:id/edit" component={UserEditPage} />
            <Route
              exact
              path="/admin/product/:id/edit"
              component={ProductEditPage}
            />
            <Route path="/shipping" component={ShippingPage} />
            <Route path="/payment" component={PaymentPage} />
            <Route path="/place-order" component={PlaceOrderPage} />
            <Route path="/order/:id" component={OrderPage} />
            <Route component={NotFound} />
          </Switch>
        </main>
        <Footer />
      </Router>
    </div>
  );
};

export default App;
