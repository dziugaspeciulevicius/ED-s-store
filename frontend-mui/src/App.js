import "./App.css";
import { createTheme } from "@mui/material";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import HomePage from "./pages/HomePage.component";
import { ThemeProvider } from "@emotion/react";
import Navbar from "./components/Navbar";
import { theme } from "./theme";

const App = () => {
  return (
    <ThemeProvider theme={theme}>
      <div>
        <Router>
          {/*<ScrollToTop />*/}
          <Navbar />
          {/*<main className="pb-3">*/}
          <main>
            <Switch>
              <Route exact path="/" component={HomePage} />
              {/*<Route exact path="/page/:pageNumber" component={HomePage} />*/}
              {/*<Route*/}
              {/*  exact*/}
              {/*  path="/search/:keyword/page/:pageNumber"*/}
              {/*  component={HomePage}*/}
              {/*/>*/}
              {/*<Route exact path="/search/:keyword" component={HomePage} />*/}
              {/*<Route path="/product/:id" component={ProductPage} />*/}
              {/*<Route path="/cart/:id?" component={CartPage} />*/}
              {/*<Route path="/shop" component={ShopPage} />*/}
              {/*<Route path="/contact" component={ContactPage} />*/}
              {/*<Route path="/about" component={AboutPage} />*/}
              {/*<Route path="/login" component={LoginPage} />*/}
              {/*<Route path="/register" component={RegisterPage} />*/}
              {/*<Route path="/profile" component={DashboardPage} />*/}
              {/*<Route exact path="/admin" component={AdminPanel} />*/}
              {/*<Route*/}
              {/*  exact*/}
              {/*  path="/admin/user/:id/edit"*/}
              {/*  component={UserEditPage}*/}
              {/*/>*/}
              {/*<Route*/}
              {/*  exact*/}
              {/*  path="/admin/product/:id/edit"*/}
              {/*  component={ProductEditPage}*/}
              {/*/>*/}
              {/*<Route path="/shipping" component={ShippingPage} />*/}
              {/*<Route path="/payment" component={PaymentPage} />*/}
              {/*<Route path="/place-order" component={PlaceOrderPage} />*/}
              {/*<Route path="/order/:id" component={OrderPage} />*/}
              {/*<Route component={NotFound} />*/}
            </Switch>
          </main>
          {/*<Footer />*/}
        </Router>
      </div>
    </ThemeProvider>
  );
};

export default App;
