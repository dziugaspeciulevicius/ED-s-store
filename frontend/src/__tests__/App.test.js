import React from "react";
import App from "../App";
import { shallow, mount, render } from "enzyme";
import "../setupTests";
import { MemoryRouter } from "react-router-dom";

import HomePage from "../pages/HomePage.component";
import NotFound from "../components/NotFound.component";
jest.mock("../pages/HomePage.component.jsx");
jest.mock("../components/NotFound.component.jsx");


describe("<App />", () => {
  it("Renders <App /> component correctly", () => {
    // shallow creates instance of a component
    expect(shallow(<App />)).toMatchSnapshot();
  });

  // it("Invalid path should redirect to 404 - Not found", () => {
  //   const wrapper1 = mount(
  //       <MemoryRouter initialEntries={["/esxrdcfvgbhnj"]}>
  //         <App />
  //       </MemoryRouter>
  //   );
  //   expect(wrapper1.find(HomePage)).toHaveLength(0);
  //   expect(wrapper1.find(NotFound)).toHaveLength(1);
  // });
});

describe("<HomePage />", () => {
  it("Renders <HomePage /> component correctly", () => {
    // shallow creates instance of a component
    expect(shallow(<HomePage />)).toMatchSnapshot();
  });
});
