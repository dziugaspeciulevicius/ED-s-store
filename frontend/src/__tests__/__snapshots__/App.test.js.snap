// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> Renders <App /> component correctly 1`] = `ShallowWrapper {}`;

exports[`<HomePage /> Renders <HomePage /> component correctly 1`] = `ShallowWrapper {}`;
